DoNotHideBaseClassMethods shouldn't apply to constructors

#### Analyzer package

Microsoft.CodeQuality.Analyzers, version [2.6.0](https://www.nuget.org/packages/Microsoft.CodeQuality.Analyzers/2.6.0)

#### Analyzer

DoNotHideBaseClassMethods

#### Repro steps

1. `dotnet new classlib`
2. `dotnet add package Microsoft.CodeQuality.Analyzers`
3. Change Class1.cs to:

    ```c#
    class Base
    {
        public Base(string input) { }
    }

    class Derived : Base
    {
        public Derived(object input) : base(null) { }
    }
    ```
4. `dotnet build`

#### Expected behavior

No CA1061, since constructors are not inherited. In other words, if I removed the `Derived` constructor, I still wouldn't be able to write `new Derived("some string")`.

#### Actual behavior

```
> dotnet build
Microsoft (R) Build Engine version 15.5.180.51428 for .NET Core
Copyright (C) Microsoft Corporation. All rights reserved.

  Restoring packages for C:\code\tmp\hwapp\hwapp.csproj...
  Generating MSBuild file C:\code\tmp\hwapp\obj\hwapp.csproj.nuget.g.props.
  Restore completed in 751,09 ms for C:\code\tmp\hwapp\hwapp.csproj.
Class1.cs(8,12): warning CA1061: Change or remove 'Derived.Derived(object)' because it hides a more specific base class method: 'Base.Base(string)'. [C:\code\tmp\hwapp\hwapp.csproj]
Class1.cs(6,7): warning CA1812: Derived is an internal class that is apparently never instantiated. If so, remove the code from the assembly. If this class is intended to contain only static members, make
it static (Shared in Visual Basic). [C:\code\tmp\hwapp\hwapp.csproj]
Class1.cs(8,27): warning CA1801: Parameter input of method .ctor is never used. Remove the parameter or use it in the method body. [C:\code\tmp\hwapp\hwapp.csproj]
Class1.cs(3,24): warning CA1801: Parameter input of method .ctor is never used. Remove the parameter or use it in the method body. [C:\code\tmp\hwapp\hwapp.csproj]
  hwapp -> C:\code\tmp\hwapp\bin\Debug\netstandard2.0\hwapp.dll

Build succeeded.

Class1.cs(8,12): warning CA1061: Change or remove 'Derived.Derived(object)' because it hides a more specific base class method: 'Base.Base(string)'. [C:\code\tmp\hwapp\hwapp.csproj]
Class1.cs(6,7): warning CA1812: Derived is an internal class that is apparently never instantiated. If so, remove the code from the assembly. If this class is intended to contain only static members, make
it static (Shared in Visual Basic). [C:\code\tmp\hwapp\hwapp.csproj]
Class1.cs(8,27): warning CA1801: Parameter input of method .ctor is never used. Remove the parameter or use it in the method body. [C:\code\tmp\hwapp\hwapp.csproj]
Class1.cs(3,24): warning CA1801: Parameter input of method .ctor is never used. Remove the parameter or use it in the method body. [C:\code\tmp\hwapp\hwapp.csproj]
    4 Warning(s)
    0 Error(s)

Time Elapsed 00:00:11.72
```

--- a/src/Microsoft.CodeQuality.Analyzers/Core/ApiDesignGuidelines/DoNotHideBaseClassMethods.cs
+++ b/src/Microsoft.CodeQuality.Analyzers/Core/ApiDesignGuidelines/DoNotHideBaseClassMethods.cs
@@ -48,7 +48,7 @@ namespace Microsoft.CodeQuality.Analyzers.ApiDesignGuidelines
             var method = (IMethodSymbol)context.Symbol;
 
             // Bail out if this method is a constructor
-            if(method.IsConstructor())
+            if (method.IsConstructor())
             {
                 return;
             }

--- a/src/Microsoft.CodeQuality.Analyzers/Core/ApiDesignGuidelines/DoNotHideBaseClassMethods.cs
+++ b/src/Microsoft.CodeQuality.Analyzers/Core/ApiDesignGuidelines/DoNotHideBaseClassMethods.cs
@@ -47,6 +47,12 @@ namespace Microsoft.CodeQuality.Analyzers.ApiDesignGuidelines
         {
             var method = (IMethodSymbol)context.Symbol;
 
+            // Bail out if this method is a constructor
+            if(method.IsConstructor())
+            {
+                return;
+            }
+
             // Bail out if this method overrides another (parameter types must match) or doesn't have any parameters
             if (method.IsOverride || !method.Parameters.Any())
             {

--- a/src/Microsoft.CodeQuality.Analyzers/UnitTests/ApiDesignGuidelines/DoNotHideBaseClassMethodsTests.cs
+++ b/src/Microsoft.CodeQuality.Analyzers/UnitTests/ApiDesignGuidelines/DoNotHideBaseClassMethodsTests.cs
@@ -118,6 +118,26 @@ End Class",
         }
 
         [Fact]
+        public void CA1061_ConstructorCallsBaseConstructorWithDifferentParameterType_NoDiagnostic()
+        {
+            VerifyCSharp(@"
+class Base
+{
+    public Base(string input)
+    {
+    }
+}
+class Derived : Base
+{
+    public Derived(object input)
+        :base(null)
+    {
+    }
+}
+");
+        }
+
+        [Fact]
         public void CA1061_DerivedMethodHasLessDerivedParameter_MultipleMethodsHidden_Diagnostics()
         {
             VerifyCSharp(@"

